File: Python/Main2.py
import tkinter as tk
import math

class Punto:
    def __init__(self, x, y):
        self.x = x
        self.y = y

    def get_x(self):
        return self.x

    def get_y(self):
        return self.y

    def __str__(self):
        return f"({self.x}, {self.y})"
    
    def coord_cartesianas(self):
        return self.x, self.y
    
    def coord_polares(self):
        radio = math.sqrt(self.x * self.x + self.y * self.y)
        angulo = math.atan(self.y / self.x)
        angulo = math.degrees(angulo)
        return radio, angulo

class Linea:
    def __init__(self, p1, p2):
        self.p1 = p1
        self.p2 = p2

    def __str__(self):
        return f"Linea [P1={self.p1}, P2={self.p2}]"

    def dibuja_linea(self, canvas):
        canvas_width = int(canvas['width'])
        canvas_height = int(canvas['height'])
        
        x_offset = canvas_width // 2
        y_offset = canvas_height // 2

        canvas.create_line(
            self.p1.get_x() + x_offset, y_offset - self.p1.get_y(),
            self.p2.get_x() + x_offset, y_offset - self.p2.get_y(),
            fill="black"
        )

class LineaCanvas(tk.Canvas):
    def __init__(self, parent, linea):
        super().__init__(parent, width=400, height=400)
        self.linea = linea
        self.update_linea()

    def update_linea(self):
        self.delete("all")
        self.linea.dibuja_linea(self)

def main():
    root = tk.Tk()
    root.title("Dibujar Linea")

    punto1 = Punto(0, 0) 
    punto2 = Punto(50, 50)  
    linea = Linea(punto1, punto2)
    canvas = LineaCanvas(root, linea)
    canvas.pack(fill=tk.BOTH, expand=True)

    root.geometry("400x400")
    root.mainloop()

if __name__ == "__main__":
    main()

File: Java/Main2.java
package practicas;

import javax.swing.*;
import java.awt.*;

class Punto {
    private int x;
    private int y;

    public Punto(int x, int y) {
        this.x = x;
        this.y = y;
    }

    public int getX() {
        return x;
    }

    public int getY() {
        return y;
    }

    @Override
    public String toString() {
        return "(" + x + ", " + y + ")";
    }
    
    public int[] coord_cartesianas() {
        return new int[] {this.x, this.y};
    }
    
    public int[] coord_polares() {
        int radio = (int)Math.sqrt(this.x * this.x + this.y * this.y);
        int angulo = (int) Math.atan((double)this.y / this.x);
        angulo = (int)Math.toDegrees(angulo);
        return new int[] {radio, angulo};
    }
}

class Linea {
    private Punto p1;
    private Punto p2;

    public Linea(Punto p1, Punto p2) {
        this.p1 = p1;
        this.p2 = p2;
    }

    @Override
    public String toString() {
        return "Linea [P1=" + p1.toString() + ", P2=" + p2.toString() + "]";
    }

    public void dibujaLinea(Graphics g, int width, int height) {
        int x_offset = width / 2;
        int y_offset = height / 2;
        
        g.drawLine(p1.getX() + x_offset, y_offset - p1.getY(), p2.getX() + x_offset, y_offset - p2.getY());
    }
}

class LineaPanel extends JPanel {
    private Linea linea;

    public LineaPanel(Linea linea) {
        this.linea = linea;
    }

    @Override
    protected void paintComponent(Graphics g) {
        super.paintComponent(g);
        int width = getWidth();
        int height = getHeight();
        linea.dibujaLinea(g, width, height);
    }
}

public class Main2 {
    public static void main(String[] args) {
        Punto punto1 = new Punto(0, 0); 
        Punto punto2 = new Punto(50, 50); 

        Linea linea = new Linea(punto1, punto2);

        JFrame frame = new JFrame("Dibujar Linea");
        LineaPanel panel = new LineaPanel(linea);

        frame.add(panel);
        frame.setSize(400, 400);
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        frame.setVisible(true);
    }
}



